companion/utils/graphql2smip.py

```diff
@@ -0,0 +1,160 @@
+import os, argparse, json, requests
+from datetime import datetime
+import json
+import pandas as pd
+import config.smip as config  #copy config-example.py to config.py and set values
+
+
+
+class graphql2smip():
+    def __init__(self,variable):
+        self.tagid =config.smip["tagids"]
+        self.authenticator=config.smip["authenticator"]
+        self.pswd=config.smip["password"]
+        self.name=config.smip["name"]
+        self.role=config.smip["role"]
+        self.url=config.smip["url"]
+        self.bearer_token=config.smip["bearer_token"]
+        self.variable=variable
+
+    def do_split(self,full_df, column_no):
+        df_new = full_df.iloc[:, [0, column_no]].copy()
+        df_new['status'] = pd.Series([1 for x in range(len(df_new.index))])
+        df_new = df_new.astype(str)
+
+        return df_new
+
+
+    def perform_graphql_request(self,content, headers=None):
+        #print("Performing request with content: ")
+        #print(content)
+
+        r = requests.post(self.url, headers=headers, data={"query": content})
+        r.raise_for_status()
+        return r.json()
+
+
+    def get_bearer_token(self):
+        response = self.perform_graphql_request(f"""
+        mutation authRequest {{
+            authenticationRequest(
+            input: {{authenticator: "{self.authenticator}", role: "{self.role}", userName: "{self.name}"}}
+            ) {{
+            jwtRequest {{
+                challenge, message
+            }}
+            }}
+        }}
+        """)
+        jwt_request = response['data']['authenticationRequest']['jwtRequest']
```

Comment:
Make sure to check that response contains data before accessing it (this can cause runtime errors if the request fails). You can do something like the following:
```
jwt_request = {}
if response:
    jwt_request = response.get("data",{}).get("authenticationRequest",{}).get("jwtRequest",{})
```
Which makes sure that if request is invalid